### PR TITLE
Version 0.12.3

### PR DESCRIPTION
**Pull request by florimondmanca**
We've got a bunch of fixes in `master` ready for a patch release. :-)

Changelog source: https://github.com/encode/httpcore/compare/0.12.2...cf45ea7

### Fixed

- Abort SSL connections on close rather than waiting for remote EOF when using `asyncio`.  (Pull #167)
- Fix exception raised in case of connect timeouts when using the `anyio` backend. (Pull #236)
- Fix `Host` header precedence for `:authority` in HTTP/2. (Pull #241, #243)
- Handle extra edge case when detecting for socket readability when using `asyncio`. (Pull #242, #244)
- Fix `asyncio` SSL warning when using proxy tunneling. (Pull #249)

### CODE REVIEW

**Review by bdraco — APPROVED** [review]

**Comment by bdraco** [comment]
@florimondmanca LGTM
https://github.com/encode/httpx/issues/859#issuecomment-757929038

Can we get a release so we can close out https://github.com/home-assistant/core/issues/43576?

**Comment by florimondmanca** [comment]
OK, lemme move this on.

**Comment by florimondmanca** [comment]
I can't by pass merge checks — need one approval from the Maintainers team. :-)

@encode/maintainers Hi all — can I get an approval on this release PR to unblock folks awaiting for the fixes? Thanks!

**Review by euri10 — APPROVED** [review]

**Comment by euri10** [comment]
> The PR is up to date, but I can't bypass merge checks — need one approval from the Maintainers team. :-)
> 
> @encode/maintainers Hi all — can I get an approval on this release PR to unblock folks awaiting for the fixes? Thanks!

here it is ! :)

**Comment by bdraco** [comment]
Thanks

**Comment by florimondmanca** [comment]
Hmm interesting, looks like the master build is broken, most likely due to some dependency drift...? https://github.com/encode/httpcore/runs/1785395873

0.12.3 was released just fine, any case. :)

**Pull request merged**
